### PR TITLE
Add features_to_explain to shap config

### PR DESCRIPTION
**Pull request by oyangz**
*Issue #, if available:* https://issues.amazon.com/issues/RAI-5317

*Description of changes:* Add features_to_explain to Clarify SHAPConfig

*Testing done:* tox tests passed

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/clarify.py [hunk 1]
```diff
@@ -1290,6 +1295,13 @@ def __init__(
                 )
         if image_config:
             _set(image_config.get_image_config(), "image_config", self.shap_config)
+        if features_to_explain is not None and (
+            text_config is not None or image_config is not None
+        ):
+            raise ValueError(
```
Comment (mufaddal-rohawala) [inline src/sagemaker/clarify.py:1301]: please add this in method docstring

**Review by mufaddal-rohawala — CHANGES_REQUESTED** [review]

**Review by malhotra18 — APPROVED** [review]
LGTM!

File: src/sagemaker/clarify.py (hunk 1 above)
Comment (oyangz) [reply src/sagemaker/clarify.py:1301]: Sure, will do.

Comment (oyangz) [reply src/sagemaker/clarify.py:1306]: Added in latest commit.

**Review by mufaddal-rohawala — APPROVED** [review]

**Comment by mufaddal-rohawala** [comment]
Bypassing notebook tests (due to a known timeout issue), the timed out notebooks have been verified successfully atleast once in subsequent iterations.

**Pull request merged**